httpx/_compat.py

```diff
@@ -17,6 +20,24 @@
     except ImportError:
         brotli = None
 
+# Zstandard support is optional
+zstd: Optional[ModuleType] = None
+try:
+    import zstandard as zstd
+except (AttributeError, ImportError, ValueError):  # Defensive:
+    zstd = None
+else:
+    # The package 'zstandard' added the 'eof' property starting
+    # in v0.18.0 which we require to ensure a complete and
+    # valid zstd stream was fed into the ZstdDecoder.
+    # See: https://github.com/urllib3/urllib3/pull/2624
+    _zstd_version = tuple(
+        map(int, re.search(r"^([0-9]+)\.([0-9]+)", zstd.__version__).groups())  # type: ignore[union-attr]
+    )
+    if _zstd_version < (0, 18):  # Defensive:
+        zstd = None
```

Comment:
This might be overly defensive? Can we pin >= 0.18 in the requirements instead?

Reply:
I asked a similar question already and got this answer:

https://github.com/encode/httpx/pull/3139#discussion_r1518815892

> zstandard is an optional install. So if your `requirements.txt` just has `httpx` and not `httpx[zstd]`, and you have `zstandard==[old version]` somehow, this would prevent breakage.

Basically the idea is that the optional dependency pin is optional and does not guarantee that the given dependency version will be installed. This check would only be redundant if we made zstandard a hard dependency (one not requiring httpx[zstandard]).

The same check [exists](https://github.com/urllib3/urllib3/blob/733f638a2faa02b4ff8a9f3b5668949d39396b8b/src/urllib3/response.py#L28-L43) in urllib3.

Personally, I +1 the extra safety at the cost of minimal startup delay. :slightly_smiling_face:

Reply:
Okay yep. Main thing is coverage needs addressing.